js/content/store.js

```diff
@@ -912,6 +913,49 @@ class AppPageClass extends StorePageClass {
         });
     }
 
+    addFullscreenScreenshotView() {
+        function es_toggleFullScreen(event) {
+            if (!document.fullscreenElement) {
+                let element = event.target.closest(".screenshot_popup_modal_content");
+                element.requestFullscreen();
+            } else {
+                document.exitFullscreen();
+            }
+        }
+
+        function es_initFSVButton(event) {
+            if (event.animationName !== "es_screenshot_popup_modal_hook") return;
+
+            let modalFooter = document.querySelector(".screenshot_popup_modal_footer");
+            let fsvButton = document.createElement('div');
+            fsvButton.classList.add("btnv6_blue_hoverfade", "btn_medium", "es_screenshot_fullscreen_toggle");
+            fsvButton.innerHTML = "<i></i>";
+            let nextButton = modalFooter.querySelector(".next");
+            let nextButtonOffsetWidth = nextButton.offsetWidth;
+            if (nextButton.style.display === "none") {
+                nextButton.style.display = "";
+                nextButtonOffsetWidth = nextButton.offsetWidth;
+                nextButton.style.display = "none";
+            }
+            fsvButton.style.right = `calc(${nextButtonOffsetWidth}px + 0.5em)`;
+            fsvButton.addEventListener("click", es_toggleFullScreen);
+            modalFooter.appendChild(fsvButton);
+
+            let downloadButton = document.createElement('a');
+            downloadButton.classList.add("btnv6_blue_hoverfade", "btn_medium", "es_screenshot_download_btn");
+            let modalTitleLink = modalFooter.parentElement.querySelector(".screenshot_popup_modal_title > a");
+            downloadButton.title = modalTitleLink.textContent.trim();
+            downloadButton.innerHTML = "<i></i>";
+            downloadButton.style.right = `calc(${nextButtonOffsetWidth + fsvButton.offsetWidth}px + 1em)`;
+            downloadButton.addEventListener("click", () => {
+                modalTitleLink.click();
+            });
+            modalFooter.appendChild(downloadButton);
+        }
+
+        document.addEventListener("animationstart", es_initFSVButton);
```

Comment:
Can't we just listen to `body`'s changes with a `MutationObserver` with the parameters `{ childList: true }`?
Because the code doesn't require the modal to be fully opaque?
Correct me if I'm wrong

Reply:
Not sure what you mean with being fully opaque, but listening to an event seems less intrusive than observing the document body.

Reply:
Actually, I really don't remember why I used the hook in this case instead of the observer.

Reply:
In case of actually making overhaul of screenshot viewer there is much better way to deal with it. Replace whole screenshot modal popup by replacing old and ugly function that creates it. Here:
```HighlightPlayer.prototype.ShowScreenshotPopup = function( screenshotid )```
at
https://steamstore-a.akamaihd.net/public/javascript/gamehighlightplayer.js (:555)